### ISSUE TITLE
Unnecessary pristine check

### ISSUE THREAD

**Issue by lord-xeon** [erikras/redux-form#2380]
On lines 551 through 555 of reduxForm.js an unnecessary and inefficient call to `deepEqual` is made.
```
if (shouldResetValues) {
    values = initial
}

const pristine = deepEqual(initial, values)
```

if `shouldResetValues` is true, then we know that `values` and `initial` are the same, therefore we can short circuit and set pristine to true without having to make an expensive `deepEqual` call on those two objects.

**Comment by gustavohenke** [comment]
Hi! Thanks for the report.
A few considerations:

> On lines 551 through 555 of reduxForm.js
   
Remember that this is relative to the version you're looking at.
We may look again at this issue one day we, and when going after those lines... they're no longer at the same place they used to be, so... always mention the commit hash/tag where the problem exists!

> without having to make an expensive deepEqual call on those two objects

If you can throw some data into this discussion regarding this check, please do so.
We had perf problems in the past, and we always have room for improvements.

But I doubt that this `deepEqual` check is the main perf bottleneck of Redux Form.

> if shouldResetValues is true, then we know that values and initial are the same, therefore we can short circuit and set pristine to true

You just proposed the fix for the problem you highlighted.
As a rule of thumb for open source collaborations: you can always send a pull request when the problem doesn't need discussions; just go ahead and do it!

**Comment by lord-xeon** [comment]
I am aware of the difference in line numbers for later views/different branches of the file, that is why I included the lines in question in the comment itself.

I tried to submit a pull request, but I kept getting the "you don't have permissions to submit a pull request to this repository" error.

**Comment by erikras** [comment]
Excellent code analysis, @lord-xeon. It was trivial enough to fix. Thanks!

**Comment by erikras** [comment]
Fix released as [`v6.5.0`](https://github.com/erikras/redux-form/releases/tag/v6.5.0).
